src/io/aleph/dirigiste/Pool.java

```diff
@@ -260,9 +259,8 @@ private Map<K,Stats> updateStats() {
     }
 
     private void addObject(K key) {
-        Queue q = queue(key);
-
         _lock.lock();
+        Queue q = queue(key);
```

Comment:
Moving the global mutex fixes the issue.
However, I think we should lock only a queue....

Reply:
https://github.com/clj-commons/dirigiste/pull/37/commits/8309378f14ca0435d8d26144728e03904056f615

Reply:
I'm a little confused here. You moved the `queue()` call inside the lock, then moved it back out? What did you think it fixed? And what was the problem that made you revert it?

Reply:
The `queue()` method creates the queue if it doesn't exist (anymore).
When the call was protected by the mutex, we were sure it cannot be destroyed
just after it has been created (because no object was present on it) during the
control period. 

By locking the Queue with `_lockedQueues`, we have the same guarantees.